components/script/dom/htmliframeelement.rs

```diff
@@ -82,8 +82,8 @@ impl HTMLIFrameElement {
             if url.is_empty() {
                 None
             } else {
-                let window = window_from_node(self);
-                window.get_url().join(&url).ok()
+                let document = document_from_node(self);
+                document.base_url().join(&url).ok()
```

Comment:
Looks like no other code is referencing `document` at all, can you make this a one-liner instead?